meteor-app/imports/ui/pages/workspace-charts/component.js

```diff
@@ -34,74 +36,76 @@ export default class ChartsPage extends React.Component {
     this._bound_yearMaxStepForwardButtonOnClick = this._yearMaxStepForwardButtonOnClick.bind(this);
   }
 
+  _rangeFilterOnChange (values) {
+    const {
+      updateFilter,
+      filterMin,
+      filterMax,
+      rangeMin,
+      rangeMax,
+    } = this.props;
+
+    updateFilter(clampFilterValue(values[0], rangeMin, filterMax), clampFilterValue(values[1], filterMin, rangeMax));
+  }
+
   _rangeFilterMinOnChange (event) {
     const target = event.currentTarget;
     const {
-      updateFilterMin,
-      rangeMin,
       filterMax,
     } = this.props;
 
-    let newValue = parseInt(target.value, 10);
-    newValue = isNaN(newValue) ? rangeMin : newValue;
-    newValue = Math.max(rangeMin, newValue);
-    newValue = Math.min(newValue, filterMax);
-    updateFilterMin(newValue);
+    this._rangeFilterOnChange([target.value, filterMax]);
   }
 
   _rangeFilterMaxOnChange (event) {
     const target = event.currentTarget;
     const {
-      updateFilterMax,
       filterMin,
-      rangeMax,
     } = this.props;
 
-    let newValue = parseInt(target.value, 10);
-    newValue = isNaN(newValue) ? filterMin : newValue;
-    newValue = Math.max(filterMin, newValue);
-    newValue = Math.min(newValue, rangeMax);
-    updateFilterMax(newValue);
+    this._rangeFilterOnChange([filterMin, target.value]);
   }
 
   _yearMinStepBackButtonOnClick (/* event */) {
     const {
       filterMin,
+      filterMax,
       rangeMin,
-      updateFilterMin,
+      updateFilter,
     } = this.props;
 
-    updateFilterMin(Math.max(filterMin - 1, rangeMin));
+    updateFilter(Math.max(filterMin - 1, rangeMin), filterMax);
```

Comment:
Then there apparently is a design issue with your helpers. You are doing half of the work of another helper here.